chain/cosmos/chain_node.go

```diff
@@ -376,14 +376,35 @@ func (tn *ChainNode) FindTxs(ctx context.Context, height uint64) ([]blockdb.Tx,
 	return txs, nil
 }
 
+func removeAtIndex(s []string, index, count int) []string {
+	ret := make([]string, 0, len(s)-count)
+	ret = append(ret, s[:index]...)
+	return append(ret, s[index+count:]...)
+}
+
 // TxCommand is a helper to retrieve a full command for broadcasting a tx
 // with the chain node binary.
 func (tn *ChainNode) TxCommand(keyName string, command ...string) []string {
 	command = append([]string{"tx"}, command...)
+	var gasPrices, gasAdjustment string = tn.Chain.Config().GasPrices, fmt.Sprint(tn.Chain.Config().GasAdjustment)
+	for i := 0; i < len(command); i++ {
+		switch command[i] {
+		case "--gas-prices":
+			gasPrices = command[i+1]
+			fmt.Println("Removing command:", command[i])
+			command = removeAtIndex(command, i, 2)
```

Comment:
I know we went through this before, but thinking about it now, why don't we just only append the chain config value flags if we don't find the flags already in the list? That should be more concise than removing and re-adding.